build.sbt

```diff
@@ -152,6 +152,7 @@ lazy val `kyo-prelude` =
         .in(file("kyo-prelude"))
         .settings(
             `kyo-settings`,
+            libraryDependencies += "com.lihaoyi" %%% "pprint"        % "0.9.0",
```

Comment:
I wasn't sure if I should add the effect to `kyo-prelude` or `kyo-core`. Although it isn't a pure effect, it seems important to make it available for codebases that have only `kyo-prelude`. I moved the `pprint` to `kyo-prelude` to use it when outputting the debugging info.

Reply:
Can you me why we need pprint?

Reply:
It's used here to print the values in the debug log. It inspects the types and outputs their structure with ansi colors instead of calling `toString`. It uses reflection for that. Example:

![image](https://github.com/user-attachments/assets/6efd9700-9c7e-4171-94ba-8dfde7255be4)